### PR TITLE
🐛 fix: Fixed class name for floating images

### PR DESCRIPTION
**Pull request by ashleytowner**
# Pull Request Overview
This fixes padding for floating images, due to incorrect class name & incorrect use of a function.

In notion:
https://www.notion.so/greenpeaceaustraliapacific/P4-Float-images-correctly-9ca9b50751554a06a26b746b1bbcdf43

## 🏁 PR Checklist

Reviewer checklist:
- [ ] ✨ This PR includes new user-facing feature and/or settings
- [ ] 📝 Documentation for new features/settings are in PR description (to go to wiki or Coral docs) 

Contributor checklist:
- [ ] Reference a GitHub issue, Jira ticket, or Notion card in the description 💯
- [ ] 📝 Update the CHANGELOG.md (Put under `[Unreleased]`). Include link to this PR.
- [ ] 📝 Add JSDoc code documentation
- [ ] 🏷 Add flow types. (If needed.)
- [ ] ✅ Add unit tests (`*.test.js`) and/or Cypress Tests.
- [ ] 📖 Create Storybook stories. (For any applicable Components.)
- [ ] 📸 Update snapshots.  Run `yarn test:watch`, press 'a' for "all tests", and "u" to update snaps if needed.
- [ ] Run tests (`yarn test` and `yarn flow`) and clean up any errors 🚀
- [x] Select 1-2 Reviewers
- [x] Feel good about yourself, you're doing great! 🥳

### CODE REVIEW

**Review by techieshark — COMMENTED** [review]
On the one hand, this does fix the immediate issue - I can see that images become aligned left.

On the other hand, I wondered where the `space(1)` was coming from, then dug up where it comes from in the parent them, and then realised a couple things: 1) to avoid technical debt and drift, can we figure out how to make use of the parent-defined variables in the child theme? and 2) why is this not being 'fixed' in the parent theme - is it not an issue there?

image


Meta comment:

1. ideally we link back from the pull request to the notion card.  I wonder what a systematic way to remember to do that would be.

2. Also I wonder if the steps to reproduce could be simpler and/or if we could have a 'steps to test' type thing. Like, it'd be great if the reviewer only needed to view an existing page in order to see the issue, then checkout the branch, and see that it's been fixed.

**Comment by ashleytowner** [comment]
@techieshark I actually spoke to Nikos about this, and he said that SCSS variables in the parent theme aren't designed to be usable in child themes, they use CSS Custom properties to achieve that when necessary. 

We *can* achieve it by using `@import` (or maybe `@use`, but I'm not sure if our version of sass supports it), so let me know if you want me to go ahead with that, but if we have to use import [it is not recommended](https://sass-lang.com/documentation/at-rules/import/)

---

As for fixing it in the master theme, I'm happy to do that, I was just under the impression that the idea was to try to fix it in the child theme if possible.

**Comment by techieshark** [comment]
@ashleytowner thanks for that context...

IMHO, yes definitely if there's a 'bug' in the parent them we should fix that. 

Or if we're implementing a 'feature' but you feel it would be broadly beneficial (to any other office using the parent them), I think we should also do it in the parent.

If neither of those apply, finding some way to access the parent would be cool and I feel like this use case (no pun intended) will come up again before long. Maybe `@use` would work, or maybe the parent theme could be updated so their compiled spacing values get emitted as CSS vars we could then use. 

But if you look into all this and it feels not easy enough, maybe we just go with what you have and add another card/issue for coming up with a better system later.

**Comment by ashleytowner** [comment]
@techieshark I've added a commit to add in SCSS support, and then use CSS variables.

As part of this it's generated a new map file, and I had to make a change to a class name in another file in order to get the SASS to compile

**Review by techieshark — APPROVED** [review]
LGTM - we just need to document the fact that people will need the version of the master theme with the assets/src dir in it!

File: style.css [hunk 1]
```diff
@@ -119,66 +126,63 @@ h3,
...
+  -webkit-transition: height 0.4s;
+  transition: height 0.4s;
+  /* bump height to get big logo (bumped up by fraction of golden ratio) */
+  height: 34.03439px;
+}
 
-body.scrolled.home .site-logo img {
-        height: 26px; /* default P4 value */
-      }
+.home .site-logo img body.scrolled.home .site-logo img {
```
Comment (techieshark) [inline style.css:183]: This doesn't seem right does it?

image

File: style/header.scss [hunk 2]
```diff
@@ -57,7 +57,7 @@
       transition: height 0.4s;
       /* bump height to get big logo (bumped up by fraction of golden ratio) */
       height: calc(26px + 1/2 * calc(1.61803 * 26px - 26px));
-      body.scrolled& {
+      body.scrolled.home .site-logo img {
```
Comment (techieshark) [inline style/header.scss:60]: I left a comment in the generated .css file but flagging this as the source of the issue. 

See https://sass-lang.com/playground/#eJwzNNTLyM9NVajmUlDQK84sSdXNyU/PB3MVFDJz06EsBYWk/JRKveLkovycnNQUBTW4OFAmMTk7vSi/NC/FSiEppzTVGipTywUja7lqFQAHBRvf 

I think you just want a space and then the ampersand.

**Review by techieshark — CHANGES_REQUESTED** [review]
The extra doc is good! Thanks for thinking of that.

Sidenote, I started a chat with P4 dev team about streamlining the install so in future we can remove the doc note protecting us from the foot gun. https://greenpeace.slack.com/archives/C0151L0KKNX/p1709120542770239

Added 1 fix suggestion to what you have.

But stepping back a sec - can you clarify, is there or is there not a defect in the parent them that could be fixed?

File: style/header.scss (hunk 2 above)
Comment (ashleytowner) [reply style/header.scss:60]: So the ampersand is what was there already, but ampersand at the end isn't supported by the version of SASS that we are using so I just replaced it with the parent.

But you're right, the output CSS doesn't seem right 🤔

**Comment by ashleytowner** [comment]
> The extra doc is good! Thanks for thinking of that.
> 
> Sidenote, I started a chat with P4 dev team about streamlining the install so in future we can remove the doc note protecting us from the foot gun. https://greenpeace.slack.com/archives/C0151L0KKNX/p1709120542770239
> 
> Added 1 fix suggestion to what you have.
> 
> But stepping back a sec - can you clarify, is there or is there not a defect in the parent them that could be fixed?

I think there *is* a defect in the parent theme, however given the timeline I was just trying to fix it in here so we can get it all through. 

Happy to make a PR for this in the parent theme though, but I'd like to see this merged, then do that separately and revert this. Do you think this is a bad approach?

File: style/header.scss (hunk 2 above)
Comment (ashleytowner) [reply style/header.scss:60]: Oh wait I see the issue!

So when you include an ampersand it raises the rule up a level, if that makes sense. 

So this
```scss
.foo {
    font-weight: bold;
    .bar {
        color: red;
        
        .baz {
            font-size: 3rem;
        }
    }
}
```

becomes this

```css
.foo {
  font-weight: bold;
}
.foo .bar {
  color: red;
}
.foo .bar .baz {
  font-size: 3rem;
}
```

But this
```scss
.foo {
    font-weight: bold;
    .bar {
        color: red;
        
        &.baz {
            font-size: 3rem;
        }
    }
}
```

becomes this:
```css
.foo {
  font-weight: bold;
}
.foo .bar {
  color: red;
}
.foo .bar.baz {
  font-size: 3rem;
}
```

*not* this:

```css
.foo {
  font-weight: bold;
}
.foo .bar {
  color: red;
}
.foo .bar .foo .bar.baz {
  font-size: 3rem;
}
```

Comment (techieshark) [reply style/header.scss:60]: @ashleytowner also what we had before had NO space before the ampersand, and when I tried in the playground I got something working by adding a space.

If we need to get a newer version of sass I'm fine with that too

File: style.css (hunk 1 above)
Comment (ashleytowner) [reply style.css]: Okay, I've fixed this now!

**Comment by techieshark** [comment]
@ashleytowner yeah for expediency I'm happy with your approach. 

I think we're on the same page - 

1. file bug report https://planet4.greenpeace.org/contribute/report-bugs/
2. make a PR upstream that fixes this
3. once merged upstream we can remove any redundant CSS we've added here (but keep the other good work like the sass package updates etc)

**Review by techieshark — APPROVED** [review]
LGTM

Bummer that Sass doesn't do the magic we can do with postcss-nested exactly. Oh well!

**Pull request merged**